workbench/test/test_scenarios.py

```diff
@@ -1,61 +1,65 @@
 """Test that all scenarios render successfully."""
 
-import lxml.html
+import unittest
 
+import ddt
 from django.test.client import Client
+import lxml.html
+from django.core.urlresolvers import reverse
 
 from xblock.core import XBlock
 from xblock.test.tools import assert_equals
+
 from workbench import scenarios
+from workbench.runtime import reset_global_state
+
+
+@ddt.ddt
+class ScenarioTest(unittest.TestCase):
+    """Test the scenario support."""
+    def setUp(self):
+        super(ScenarioTest, self).setUp()
+        reset_global_state()
+
+    def test_all_scenarios(self):
+        """Load the home page, examine the scenarios displayed."""
+        client = Client()
+        response = client.get("/")
+        assert response.status_code == 200
+        html = lxml.html.fromstring(response.content)
+        a_tags = list(html.xpath('//a'))
+
+        # Load the loaded_scenarios from the classes.
+        loaded_scenarios = scenarios.SCENARIOS.values()
+
+        # We should have an <a> tag for each scenario.
+        assert_equals(len(a_tags), len(loaded_scenarios))
+
+        # We should have at least one scenario with a vertical tag, since we use
+        # empty verticals as our canary in the coal mine that something has gone
+        # horribly wrong with loading the loaded_scenarios.
+        assert any("<vertical_demo>" in scen.xml for scen in loaded_scenarios)
+
+        # Since we are claiming in try_scenario that no vertical is empty, let's
+        # eliminate the possibility that a scenario has an actual empty vertical.
+        assert all("<vertical_demo></vertical_demo>" not in scen.xml for scen in loaded_scenarios)
+        assert all("<vertical_demo/>" not in scen.xml for scen in loaded_scenarios)
 
+    @ddt.data(*scenarios.SCENARIOS.keys())
+    def test_scenario(self, scenario_id):
+        # This is a very shallow test.  We don't know enough about each scenario to
```

Comment:
Old habit from nose printing docstrings instead of test names when running tests.  I can add them.